datafusion/physical-plan/src/sorts/sort.rs

```diff
@@ -921,7 +922,13 @@ impl ExecutionPlan for SortExec {
     }
 
     fn statistics(&self) -> Result<Statistics> {
-        self.input.statistics()
+        statistics_with_fetch(
```

Comment:
The fact that another execution plan uses `statistics_with_fetch` is more reason in my mind to  ut `with_fetch` on `Statistics` itself (could totally be done as a follow on PR)